Pinpoint.Core/PluginSettings.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Pinpoint.Core
+{
+    public class PluginSettings : ObservableCollection<PluginSetting>
+    {
+        private readonly Dictionary<string, PluginSetting> _settingsByName = new Dictionary<string, PluginSetting>();
+
+        public void Put(string name, object value)
+        {
+            Add(_settingsByName[name] = new PluginSetting
+            {
+                Name = name,
+                Value = value
+            });
+        }
+
+        public object this[string name]
+        {
+            get => _settingsByName.GetValueOrDefault(name, null)?.Value;
+            set => Put(name, value);
+        }
+
+        public bool Has(string name) => _settingsByName.ContainsKey(name);
+        
+        public bool Remove(string name)
+        {
+            if (!_settingsByName.ContainsKey(name))
+            {
+                return false;
+            }
+
+            Remove(_settingsByName[name]);
+            return _settingsByName.Remove(name);
+        }
+
+        public string Str(string name) => this[name]?.ToString();
+
+        public int Int(string name) => int.Parse(this[name]?.ToString() ?? "0");
```

Comment:
This will throw an exception in the case where this[name] is a not-null string that cannot be parsed as an int. You could do:

```
public int Int(string name) 
{
    int.TryParse(this[name]?.ToString(), out var value);
    return value;
}
```